crates/type-system/src/ontology/uri/mod.rs

```diff
@@ -87,8 +87,7 @@ impl FromStr for VersionedUri {
     type Err = ParseVersionedUriError;
 
     fn from_str(uri: &str) -> Result<Self, ParseVersionedUriError> {
-        let (base_uri, version) = uri.rsplit_once("/v/").ok_or(ParseVersionedUriError)?;
-
+        let (base_uri, version) = uri.rsplit_once("v/").ok_or(ParseVersionedUriError)?;
```

Comment:
What is the reasoning behind this? Do we allow URIs like `https://example.com/serv/1`?

Reply:
That's a good point actually, and is a bug. Thank you. I will need to adjust this logic.
The issue is as follows
`http://ex.com/foo/v/1` will be split into
`["http://ex.com/foo", "1"]`
and `Url::parse("http://ex.com/foo")` gives a valid `url`, however
`url.join("/v/1")` will result in `foo` being dropped because it doesn't have a trailing `slash` which means it's treated as a file and is dropped. So you end up with `http://ex.com/v/1`

Can you think of a clean solution to this beyond me modifying the `base_uri` to append `"/"`?
We could perhaps do a regex with capturing groups but I'm not sure if that's cleaner or not

Reply:
Hmm, what about `format!("{base_uri}/v/{version}")`? Using `Path` may have weird side effects when on Windows, which could result in `{base_uri}\v\{version}`.

Reply:
I'm a little confused, we're not talking about `Path` or reconstructing the URI. We're talking about _extracting_ the components, hence the `rsplit`

Reply:
My point is that `base_uri` _must_ have a trailing slash or else the `Url` library doesn't treat it as a valid base.

Reply:
I'm using a regex now